python/samples/concepts/agents/step4_openai_assistant_agent_code_interpreter.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) Microsoft. All rights reserved.
+import asyncio
+
+from semantic_kernel.agents.open_ai.azure_assistant_agent import AzureAssistantAgent
+from semantic_kernel.agents.open_ai.open_ai_assistant_agent import OpenAIAssistantAgent
+from semantic_kernel.contents.chat_message_content import ChatMessageContent
+from semantic_kernel.contents.utils.author_role import AuthorRole
+from semantic_kernel.kernel import Kernel
+
+AGENT_NAME = "CodeRunner"
+AGENT_INSTRUCTIONS = "Run the provided code file and return the result."
+
+# Note: you may toggle this to switch between AzureOpenAI and OpenAI
+use_azure_openai = True
+
+
+# A helper method to invoke the agent with the user input
+async def invoke_agent(agent: OpenAIAssistantAgent, thread_id: str, input: str) -> None:
+    """Invoke the agent with the user input."""
+    await agent.add_chat_message(thread_id=thread_id, message=ChatMessageContent(role=AuthorRole.USER, content=input))
+
+    print(f"# {AuthorRole.USER}: '{input}'")
+
+    async for content in agent.invoke(thread_id=thread_id):
+        if content.role != AuthorRole.TOOL:
+            print(f"# {content.role}: {content.content}")
+
+
+async def main():
+    # Create the instance of the Kernel
+    kernel = Kernel()
+
+    # Add the OpenAIChatCompletion AI Service to the Kernel
+    service_id = "agent"
+
+    # Create the agent
+    if use_azure_openai:
+        agent = AzureAssistantAgent(
+            kernel=kernel,
+            service_id=service_id,
+            name=AGENT_NAME,
+            instructions=AGENT_INSTRUCTIONS,
+            enable_code_interpreter=True,
+        )
+    else:
+        agent = OpenAIAssistantAgent(
+            kernel=kernel,
+            service_id=service_id,
+            name=AGENT_NAME,
+            instructions=AGENT_INSTRUCTIONS,
+            enable_code_interpreter=True,
+        )
+
+    await agent.create_assistant(enable_code_interpreter=True)
```

Comment:
nit: Does `enable_code_interpreter=True` need to be set on the agent and passed to `create_assistant`?

Reply:
It looks like `enable_file_search` is only specified at the agent level?

Reply:
We can remove the bool flag in the create_assistant call since it is specified at the class attribute level (the create assistant bool would take precedence over the class attribute.

The class accepts both enable_file_search and enable_code_interpreter, which can be specified at the agent (assistant) level, the thread level and for a particular run. Although the thread level options are to pass in code_interpreter_file_ids / vector_store_id, which would supersede any of those that were previously configured at an assistant level.